packages/devextreme/js/__internal/core/utils/m_date.ts

```diff
@@ -234,7 +234,11 @@ function addDateInterval(value, interval, dir) {
     result.setMonth(result.getMonth() + 3 * intervalObject.quarters * dir);
   }
   if (intervalObject.months) {
-    result.setMonth(result.getMonth() + intervalObject.months * dir);
+    const targetMonth = result.getMonth() + intervalObject.months * dir;
+    result.setDate(1);
+    result.setMonth(targetMonth);
+    // eslint-disable-next-line max-len
```

Comment:
[code style]
Why don't u just fix it?

Reply:
[7458d66](https://github.com/DevExpress/DevExtreme/pull/28423/commits/7458d66f3dcb1882806a361d8f83596b6bf1ef54)